requirements/testing/travis_flake8.txt

```diff
@@ -0,0 +1,8 @@
+# Extra pip requirements for the travis flake8 build
+
+flake8
+flake8-per-file-ignores
+ipykernel
```

Comment:
Why do we need these last 4 dependencies to run flake8?

Reply:
Likely, we don't need these. I just replicated the 3.6 environment to be on the safe side. Removed now.

Am I correct, that flake8 just parses the source but does not import it? In that case we probably wouldn't need a lot of the apt packages as well for this environment. However that would require some reorganization of the `.travis.yml` config.

Reply:
Correct.

Reply:
Would there be interest in reorganizing this stuff (likely need to move some things into files in `requirements/testing`) or should I leave `.travis.yml` as is?

Reply:
I think there is interest in re-organizing, particularly if the flake8 build can be made faster. We merged to move it along, but further refinement welcome...